### PR TITLE
V3: HV: parse seed from ABL

### PR DESCRIPTION
**Pull request by YadongQi**
ABL pass seed_lists to HV through different interface/structures.
So, in this patch, add interface to retrieve seed from ABL and refactor
the seed parsing logic.

Signed-off-by: Qi Yadong <yadong.qi@intel.com>

### CODE REVIEW

**Comment by YadongQi** [comment]
@bzhu5 @kwang13
updated, please review it if you got a chance.

File: hypervisor/boot/sbl/abl_seed_parse.c [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		goto fail;
+
+	/*
+	 * The seed_list from ABL contains several seeds which based on SVN
+	 * and one legacy seed which is not based on SVN. The legacy seed's
+	 * svn is the minimum in the seed list. And CSE ensures at least two
+	 * seeds will be generated which will contain the legacy seed.
+	 * Here find the legacy seed index first.
+	 */
+	for (i = 0U; i < sec_info->num_seeds; i++) {
```
Comment (kwang13) [inline hypervisor/boot/sbl/abl_seed_parse.c]: use 1 instead of 0 here

File: hypervisor/boot/sbl/abl_seed_parse.c [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+		if (sec_info->seed_list[i].svn <
```
Comment (bingzhux) [inline hypervisor/boot/sbl/abl_seed_parse.c:51]: do you need to check sec_info->num_seeds value against the max value?

Comment (bingzhux) [reply hypervisor/boot/sbl/abl_seed_parse.c:51]: and minimal value is 2

File: hypervisor/boot/sbl/abl_seed_parse.c (hunk 1 above)
Comment (bingzhux) [reply hypervisor/boot/sbl/abl_seed_parse.c]: why 1?

Comment (YadongQi) [reply hypervisor/boot/sbl/abl_seed_parse.c]: legacy_seed_index initial value is 0, so i starts from 1 is correct.

File: hypervisor/boot/sbl/abl_seed_parse.c (hunk 2 above)
Comment (YadongQi) [reply hypervisor/boot/sbl/abl_seed_parse.c:51]: So if it is not in the range[2..MAX], what behavior should take? just goto fail?

File: hypervisor/boot/sbl/abl_seed_parse.c [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+				sec_info->seed_list[legacy_seed_index].svn) {
+			legacy_seed_index = i;
+		}
+	}
+
+	/*
+	 * Copy out abl_seed for trusty and clear the original seed in memory.
+	 * The SOS requires the legacy seed for RPMB usage. So skip the legacy
```
Comment (bingzhux) [inline hypervisor/boot/sbl/abl_seed_parse.c]: The SOS requires the legacy seed for RPMB usage -> 
The SOS requires the legacy seed to derive RPMB key

File: hypervisor/boot/sbl/multiboot.c [hunk 4]
```diff
@@ -287,21 +207,22 @@ int init_vm_boot_info(struct vm *vm)
...
-		/*
-		 * Convert ImageBootParamsAddr to SOS GPA and append to
-		 * kernel cmdline
-		 */
-		if (boot_params_addr != NULL) {
-			(void)memset(buf, 0U, sizeof(buf));
-			snprintf(buf, MAX_BOOT_PARAMS_LEN, "%s0x%X ",
-				boot_params_arg, hva2gpa(vm, boot_params_addr));
+		(void)memset(buf, 0U, sizeof(buf));
+		status = sbl_seed_parse(vm, cmd_src, buf, sizeof(buf));
```
Comment (bingzhux) [inline hypervisor/boot/sbl/multiboot.c:222]: what does this mean? can you add comments? e.g. first attempt to get from sbl, if failure, then try abl?

Comment (YadongQi) [reply hypervisor/boot/sbl/multiboot.c:222]: OK, will add comments

**Pull request closed without merging**